ui/tools/problemreporter/problemreporterwidget.cpp

```diff
@@ -0,0 +1,227 @@
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "problemreporterwidget.h"
+#include "ui_problemreporterwidget.h"
+#include "problemclientmodel.h"
+#include "problemreporterclient.h"
+
+#include <ui/searchlinecontroller.h>
+#include <ui/contextmenuextension.h>
+
+#include <common/objectbroker.h>
+#include <common/objectmodel.h>
+#include <common/tools/problemreporter/problemmodelroles.h>
+
+#include <QMenu>
+#include <QStandardItemModel>
+#include <QStyledItemDelegate>
+#include <QPainter>
+#include <QMouseEvent>
+#include <QStyleOptionViewItemV4>
+
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 5, 0)
+Q_DECLARE_METATYPE(Qt::CheckState)
+#endif
+
+using namespace GammaRay;
+
+class ProblemFilterDelegate : public QStyledItemDelegate
+{
+public:
+    explicit ProblemFilterDelegate(QAbstractItemView *view)
+     : QStyledItemDelegate(view)
+    {}
+
+protected:
+    void paint(QPainter *painter, const QStyleOptionViewItem &option,
+               const QModelIndex &index) const override
+    {
+        const QStyleOptionViewItemV4 *viewItemOption =
+           qstyleoption_cast<const QStyleOptionViewItemV4 *>(&option); // This is for Qt4 compatibility. Should be a noop in Qt5
+        QStyle *style = viewItemOption->widget ? viewItemOption->widget->style() : QApplication::style();
+        QStyleOptionButton checkboxOption = copyStyleOptions(option, index);
+        painter->save();
+        QString title = index.data(Qt::DisplayRole).toString();
+        const QString description = index.data(Qt::ToolTipRole).toString();
+        const int vMargin = viewItemOption->widget->style()->pixelMetric(QStyle::PM_FocusFrameVMargin);
+        const int hMargin = viewItemOption->widget->style()->pixelMetric(QStyle::PM_FocusFrameHMargin)
+                          + viewItemOption->widget->style()->pixelMetric(QStyle::PM_CheckBoxLabelSpacing);
+        const QRect checkboxRect = style->subElementRect(QStyle::SE_CheckBoxIndicator, &checkboxOption, viewItemOption->widget);
+        const QRect textRect = option.rect.adjusted(checkboxRect.width() + hMargin, vMargin, -hMargin, -vMargin);
+        const QFontMetrics metrics(option.font);
+        const QRect titleRect(textRect.left(),
+                              textRect.top(),
+                              textRect.width(),
+                              metrics.height());
+        const QRect descriptionRect = metrics.boundingRect(textRect, Qt::TextWordWrap, description)
```

Comment:
You're wrapping the text in a rectangle that includes the space for the title.
Plus, you're doing that work twice (once here and once during painting).

Why not just compute the available rect, and wrap during drawText?
    const QRect descriptionRect = textRect.adjust(0, metrics.height(), 0, 0);